### PR TITLE
Upload images from Trix editor

### PR DESCRIPTION
**Pull request by gorka**
This PR updates Trix and enables the _attachment_ button to publish images in your entries:

- Remove Trix files and require a newer version [from a CDN](https://cdn.jsdelivr.net).

Because Trix wraps attachments in a figure tag with custom data attributes:
- Transform the _figure_ and _figcaption_ tags into an _img_ with _alt_ tag when posting to a micropub endpoint.
- Transform all the _img_ with _alt_ tags into _figure_ with _figcaption_ tags when editing an entry with Trix.

The JS code was 99% copied from an [official Trix file](https://trix-editor.org/js/attachments.js) linked from the main [README](https://github.com/basecamp/trix/blob/main/README.md).

### CODE REVIEW

**Comment by barryf** [comment]
Thanks for the PR, @gorka. I'll try to put some time aside to review it this week.

**Review by barryf — COMMENTED** [review]
I wasn't able to get this to work, @gorka. An image was displayed inline but when submitting the form (or previewing) there was no figure tag or other HTML inserted. 

NB: when testing I didn't have a Micropub media endpoint defined. I don't know if that would make a difference but it should be supported.

How does this support [authentication][1] of the Micropub media endpoint? 

[1]: https://micropub.spec.indieweb.org/#authentication

File: views/layout.erb [hunk 1]
```diff
@@ -70,5 +70,6 @@
     <script src="/scripts/jquery.ns-autogrow.min.js"></script>
     <script src="/scripts/bootstrap-tokenfield.min.js"></script>
     <script src="/scripts/micropublish.js?v=20220312"></script>
+    <script src="/scripts/trix.js"></script>
```
[the remark this answers is not in the document]
Comment (gorka) [reply views/layout.erb:73]: The script on line 18 is Trix itself. The one on line 73 is Micropublish code that interacts with Trix.

File: views/form.erb [hunk 2]
```diff
@@ -317,7 +317,7 @@
               <textarea id="content" class="form-control" rows="5" name="content[][html]"><%= content_html_value %></textarea>
               <%= autogrow_script('content') %>
             <% else %>
-              <textarea id="content-html" class="form-control" rows="5" name="content[][html]"><%= content_html_value %></textarea>
+              <textarea id="content-html" class="form-control" rows="5" name="content[][html]" data-media-endpoint="<%= micropub.media_endpoint %>"><%= content_html_value %></textarea>
```
[the remark this answers is not in the document]
Comment (gorka) [reply views/form.erb:320]: I see. I'll use this value to determine if the attachments button should be shown or not.
Without a media endpoint this feature should be disabled.

**Comment by gorka** [comment]
Thanks for the review, @barryf. This feature needs a media endpoint.
I didn't realize about the media endpoint authentication. Thanks for pointing that out.

I'll update the PR with the fixes.

**Comment by gorka** [comment]
@barryf PR updated.

Now the request is sent with Authorization header.
Also, if there is no media endpoint, the button to attach media won't work and neither won't the drag & drop.

**Comment by barryf** [comment]
@gorka thanks for update. However, new problem is that I'm now getting CORS errors because of the `Authorization` header triggering a preflight:

    Access to XMLHttpRequest at 'http://localhost:3333/media' from origin 'http://localhost:5000' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

I'm wondering how this is working in your environment. Do you have CORS headers on your media endpoint that allows access from any origin?

**Comment by gorka** [comment]
@barryf yes, I updated the headers allowing from certain origins. The other option, I guess, is to upload the image to the micropublish server and make the request from there. I'm not sure if that will fix anything.

**Comment by gorka** [comment]
@barryf I updated the code to use the existing `/media` endpoint. Now the files will go through the server instead of the browser. Hopefully it works fine now.

**Comment by barryf** [comment]
Merged! Thanks again and apologies for the glacial PR review speed. August has been a busy month here.

**Comment by gorka** [comment]
@barryf Thank you for accepting it! I forgot to check and what a wonderful surprise to see it was merged 2 weeks ago.

**Pull request merged**